modules/auth/proc-adding-oauth2-proxy-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,83 @@
+[id='proc-adding-oauth2-proxy-as-an-authentication-provider_{context}']
+= Adding OAuth2 proxy as an authentication provider in {product}
+
+The {product} `core-plugin-api` package includes an OAuth2-proxy authentication provider
+that can authenticate users
+by using an link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2 proxy] in front of an {product} instance.
+This enables to reuse existing authentications within a cluster.
+In general, the OAuth2 proxy supports all OpenID Connect providers.
+For more details, see link:https://oauth2-proxy.github.io/oauth2-proxy/configuration/providers/[the supported provider list].
+
+.Procedure
+
+. Set up the OAuth2-proxy authentication provider in the {product} backend (`app-config.yaml`).
++
+Add the `oauth2Proxy` provider configuration to your `app-config.yaml` under the root `auth` configuration:
++
+[source,yaml]
+----
+auth:
+  environment: development
+  providers:
+    oauth2Proxy: {}
+----
++
+The OAuth2-proxy provider is a structure with no configuration keys.
+However, it requires the empty object to enable the provider in the auth backend.
+
+. To use the oauth2Proxy provider you must also configure it with a sign-in resolver.
++
+For the oauth2Proxy provider, the sign-in result is quite different than other providers.
+Because it's a proxy provider that can be configured to forward information through arbitrary headers, the auth result simply just gives you access to the HTTP headers of the incoming request.
+Using these you can either extract the information directly, or grab ID or access tokens to look up additional information and/or validate the request.
++
+A simple sign-in resolver might for example look like this:
++
+.`packages/backend/src/plugins/auth.ts`
```

Comment:
In RHDH container it is not possible to edit files like this, there needs to be another way to configure this